plugins/in_winlog/pack.c

```diff
@@ -372,7 +372,12 @@ void winlog_pack_event(msgpack_packer *mp_pck, PEVENTLOGRECORD evt,
     /* EventId */
     msgpack_pack_str(mp_pck, 7);
     msgpack_pack_str_body(mp_pck, "EventID", 7);
-    msgpack_pack_uint32(mp_pck, evt->EventID);
+    msgpack_pack_uint16(mp_pck, evt->EventID & 0xffff);
+
+    /* Qualifiers */
+    msgpack_pack_str(mp_pck, 10);
+    msgpack_pack_str_body(mp_pck, "Qualifiers", 10);
+    msgpack_pack_uint16(mp_pck, evt->EventID >> 16);
```

Comment:
Hi @fujimotos , apologies in advance since my C knowledge is more than rusty (used it in on by BS degree 15 years ago). When shifting integers having a "1" on their left side, I recall there was some kind of issue in which the "1" was then copied all over the MSBs. After a quick check, though, I saw that[ this supposedly happens when the original number is negative](https://stackoverflow.com/questions/14396858/right-shift-with-zeros-at-the-beginning). So, I guess this is fine, right? Or would it be necessary to do `(evt->EventID >> 16) && 0xffff`?

Reply:
@jsubirat As far as I can tell, the current implementation is just fine.

> When shifting integers having a "1" on their left side, I recall there was some kind of issue in which the "1" was then copied all over the MSBs

This issue occurs when we apply bit-shift on a negative integer. So
it becomes problematic only when the operatee is a signed type.

In contrast, `EventID` is declared as DWORD (which is an _unsigned_ integer
in Windows world). So  `EventID >> 16` should always work as expected.

Reply:
Thanks for the clarification! :-)